pulp_gem/specs.py

```diff
@@ -60,8 +203,30 @@ def analyse_gem(file_obj):
     """
     with TarFile(fileobj=file_obj) as archive:
         with archive.extractfile("metadata.gz") as md_file:
-            data = yaml.safe_load(gzip.decompress(md_file.read()))
+            data = yaml.load(gzip.decompress(md_file.read()), Loader=RubyMarshalYamlLoader)
+    gem_info = {
+        "name": data.values["name"],
+        "version": data.values["version"].values["version"],
+    }
+    # Sanitize name
+    if not NAME_REGEX.match(gem_info["name"]):
+        raise ValueError(f"Invalid gem name: {gem_info['name']}")
+    # Sanitize version
+    if VERSION_REGEX.match(gem_info["version"]):
+        gem_info["prerelease"] = False
+    elif PRERELEASE_VERSION_REGEX.match(gem_info["version"]):
+        gem_info["prerelease"] = True
+    else:
+        raise ValueError(f"Invalid version string: {gem_info['version']}")
+    for key in ("required_ruby_version", "required_rubygems_version"):
+        if (requirement := data.values.get(key)) is not None:
+            gem_info[key] = _collapse_requirement(requirement)
+    if (dependencies := data.values.get("dependencies")) is not None:
+        gem_info["dependencies"] = {
+            dep.values["name"]: _collapse_requirement(dep.values["requirement"])
```

Comment:
Is this correct? It seems wrong since `_collapse_requirement` seems to be expecting a dictionary with the key 'requirements'.

Reply:
Yes, i think it is. (Attaching a dump of the data object for documentation below.) 

```python
UsrMarshal:Gem::Specification({'name': 'pulpcore_client', 'version': UsrMarshal:Gem::Version({'version': '3.28.0'}), 'platform': 'ruby', 'authors': ['OpenAPI-Generator'], 'autorequire': None, 'bindir': 'bin', 'cert_chain': [], 'date': datetime.datetime(2023, 6, 14, 0, 0, tzinfo=datetime.timezone.utc), 'dependencies': [UsrMarshal:Gem::Dependency({'name': 'faraday', 'requirement': UsrMarshal:Gem::Requirement({'requirements': [['>=', UsrMarshal:Gem::Version({'version': '1.0.1'})], ['<', UsrMarshal:Gem::Version({'version': '2.0'})]]}), 'type': ':runtime', 'prerelease': False, 'version_requirements': UsrMarshal:Gem::Requirement({'requirements': [['>=', UsrMarshal:Gem::Version({'version': '1.0.1'})], ['<', UsrMarshal:Gem::Version({'version': '2.0'})]]})}), UsrMarshal:Gem::Dependency({'name': 'json', 'requirement': UsrMarshal:Gem::Requirement({'requirements': [['>=', UsrMarshal:Gem::Version({'version': '2.1.0'})], ['~>', UsrMarshal:Gem::Version({'version': '2.1'})]]}), 'type': ':runtime', 'prerelease': False, 'version_requirements': UsrMarshal:Gem::Requirement({'requirements': [['>=', UsrMarshal:Gem::Version({'version': '2.1.0'})], ['~>', UsrMarshal:Gem::Version({'version': '2.1'})]]})}), UsrMarshal:Gem::Dependency({'name': 'rspec', 'requirement': UsrMarshal:Gem::Requirement({'requirements': [['>=', UsrMarshal:Gem::Version({'version': '3.6.0'})], ['~>', UsrMarshal:Gem::Version({'version': '3.6'})]]}), 'type': ':development', 'prerelease': False, 'version_requirements': UsrMarshal:Gem::Requirement({'requirements': [['>=', UsrMarshal:Gem::Version({'version': '3.6.0'})], ['~>', UsrMarshal:Gem::Version({'version': '3.6'})]]})})], 'description': 'Fetch, Upload, Organize, and Distribute Software Packages', 'email': ['pulp-list@redhat.com'], 'executables': [], 'extensions': [], 'extra_rdoc_files': [], 'files': ['Gemfile', 'README.md', 'Rakefile', 'docs/AccessPoliciesApi.md', 'docs/AccessPolicy.md', 'docs/AccessPolicyResponse.md', 'docs/Artifact.md', 'docs/ArtifactDistributionResponse.md', 'docs/ArtifactResponse.md', 'docs/ArtifactsApi.md', 'docs/AsyncOperationResponse.md', 'docs/ContentApi.md', 'docs/ContentAppStatusResponse.md', 'docs/ContentGuardResponse.md', 'docs/ContentRedirectContentGuard.md', 'docs/ContentRedirectContentGuardResponse.md', 'docs/ContentSettingsResponse.md', 'docs/ContentSummaryResponse.md', 'docs/ContentguardsApi.md', 'docs/ContentguardsContentRedirectApi.md', 'docs/ContentguardsRbacApi.md', 'docs/DatabaseConnectionResponse.md', 'docs/DistributionResponse.md', 'docs/DistributionsApi.md', 'docs/DistributionsArtifactsApi.md', 'docs/Domain.md', 'docs/DomainResponse.md', 'docs/DomainsApi.md', 'docs/EvaluationResponse.md', 'docs/ExportersFilesystemApi.md', 'docs/ExportersFilesystemExportsApi.md', 'docs/ExportersPulpApi.md', 'docs/ExportersPulpExportsApi.md', 'docs/FilesystemExport.md', 'docs/FilesystemExportResponse.md', 'docs/FilesystemExporter.md', 'docs/FilesystemExporterResponse.md', 'docs/Group.md', 'docs/GroupProgressReportResponse.md', 'docs/GroupResponse.md', 'docs/GroupRole.md', 'docs/GroupRoleResponse.md', 'docs/GroupUser.md', 'docs/GroupUserResponse.md', 'docs/GroupsApi.md', 'docs/GroupsRolesApi.md', 'docs/GroupsUsersApi.md', 'docs/ImportResponse.md', 'docs/ImportersPulpApi.md', 'docs/ImportersPulpImportCheckApi.md', 'docs/ImportersPulpImportsApi.md', 'docs/MethodEnum.md', 'docs/MinimalTaskResponse.md', 'docs/MultipleArtifactContentResponse.md', 'docs/MyPermissionsResponse.md', 'docs/NestedRole.md', 'docs/NestedRoleResponse.md', 'docs/ObjectRolesResponse.md', 'docs/OrphansApi.md', 'docs/OrphansCleanup.md', 'docs/OrphansCleanupApi.md', 'docs/PaginatedAccessPolicyResponseList.md', 'docs/PaginatedArtifactDistributionResponseList.md', 'docs/PaginatedArtifactResponseList.md', 'docs/PaginatedContentGuardResponseList.md', 'docs/PaginatedContentRedirectContentGuardResponseList.md', 'docs/PaginatedDistributionResponseList.md', 'docs/PaginatedDomainResponseList.md', 'docs/PaginatedFilesystemExportResponseList.md', 'docs/PaginatedFilesystemExporterResponseList.md', 'docs/PaginatedGroupResponseList.md', 'docs/PaginatedGroupRoleResponseList.md', 'docs/PaginatedGroupUserResponseList.md', 'docs/PaginatedImportResponseList.md', 'docs/PaginatedMultipleArtifactContentResponseList.md', 'docs/PaginatedPublicationResponseList.md', 'docs/PaginatedPulpExportResponseList.md', 'docs/PaginatedPulpExporterResponseList.md', 'docs/PaginatedPulpImporterResponseList.md', 'docs/PaginatedRBACContentGuardResponseList.md', 'docs/PaginatedRemoteResponseList.md', 'docs/PaginatedRepositoryResponseList.md', 'docs/PaginatedRepositoryVersionResponseList.md', 'docs/PaginatedRoleResponseList.md', 'docs/PaginatedSigningServiceResponseList.md', 'docs/PaginatedTaskGroupResponseList.md', 'docs/PaginatedTaskResponseList.md', 'docs/PaginatedTaskScheduleResponseList.md', 'docs/PaginatedUploadResponseList.md', 'docs/PaginatedUpstreamPulpResponseList.md', 'docs/PaginatedUserResponseList.md', 'docs/PaginatedUserRoleResponseList.md', 'docs/PaginatedWorkerResponseList.md', 'docs/PatchedAccessPolicy.md', 'docs/PatchedContentRedirectContentGuard.md', 'docs/PatchedDomain.md', 'docs/PatchedFilesystemExporter.md', 'docs/PatchedGroup.md', 'docs/PatchedPulpExporter.md', 'docs/PatchedPulpImporter.md', 'docs/PatchedRBACContentGuard.md', 'docs/PatchedRole.md', 'docs/PatchedTaskCancel.md', 'docs/PatchedUpstreamPulp.md', 'docs/PatchedUser.md', 'docs/PolicyEnum.md', 'docs/ProgressReportResponse.md', 'docs/PublicationResponse.md', 'docs/PublicationsApi.md', 'docs/PulpExport.md', 'docs/PulpExportResponse.md', 'docs/PulpExporter.md', 'docs/PulpExporterResponse.md', 'docs/PulpImport.md', 'docs/PulpImportCheck.md', 'docs/PulpImportCheckResponse.md', 'docs/PulpImporter.md', 'docs/PulpImporterResponse.md', 'docs/Purge.md', 'docs/RBACContentGuard.md', 'docs/RBACContentGuardResponse.md', 'docs/ReclaimSpace.md', 'docs/RedisConnectionResponse.md', 'docs/RemoteResponse.md', 'docs/RemoteResponseHiddenFields.md', 'docs/RemotesApi.md', 'docs/Repair.md', 'docs/RepairApi.md', 'docs/RepositoriesApi.md', 'docs/RepositoriesReclaimSpaceApi.md', 'docs/RepositoryResponse.md', 'docs/RepositoryVersionResponse.md', 'docs/RepositoryVersionsApi.md', 'docs/Role.md', 'docs/RoleResponse.md', 'docs/RolesApi.md', 'docs/SigningServiceResponse.md', 'docs/SigningServicesApi.md', 'docs/StatesEnum.md', 'docs/StatusApi.md', 'docs/StatusResponse.md', 'docs/StorageClassEnum.md', 'docs/StorageResponse.md', 'docs/TaskGroupOperationResponse.md', 'docs/TaskGroupResponse.md', 'docs/TaskGroupsApi.md', 'docs/TaskResponse.md', 'docs/TaskScheduleResponse.md', 'docs/TaskSchedulesApi.md', 'docs/TasksApi.md', 'docs/Upload.md', 'docs/UploadChunk.md', 'docs/UploadChunkResponse.md', 'docs/UploadCommit.md', 'docs/UploadDetailResponse.md', 'docs/UploadResponse.md', 'docs/UploadsApi.md', 'docs/UpstreamPulp.md', 'docs/UpstreamPulpResponse.md', 'docs/UpstreamPulpsApi.md', 'docs/User.md', 'docs/UserGroup.md', 'docs/UserGroupResponse.md', 'docs/UserResponse.md', 'docs/UserRole.md', 'docs/UserRoleResponse.md', 'docs/UsersApi.md', 'docs/UsersRolesApi.md', 'docs/VersionResponse.md', 'docs/WorkerResponse.md', 'docs/WorkersApi.md', 'lib/pulpcore_client.rb', 'lib/pulpcore_client/api/access_policies_api.rb', 'lib/pulpcore_client/api/artifacts_api.rb', 'lib/pulpcore_client/api/content_api.rb', 'lib/pulpcore_client/api/contentguards_api.rb', 'lib/pulpcore_client/api/contentguards_content_redirect_api.rb', 'lib/pulpcore_client/api/contentguards_rbac_api.rb', 'lib/pulpcore_client/api/distributions_api.rb', 'lib/pulpcore_client/api/distributions_artifacts_api.rb', 'lib/pulpcore_client/api/domains_api.rb', 'lib/pulpcore_client/api/exporters_filesystem_api.rb', 'lib/pulpcore_client/api/exporters_filesystem_exports_api.rb', 'lib/pulpcore_client/api/exporters_pulp_api.rb', 'lib/pulpcore_client/api/exporters_pulp_exports_api.rb', 'lib/pulpcore_client/api/groups_api.rb', 'lib/pulpcore_client/api/groups_roles_api.rb', 'lib/pulpcore_client/api/groups_users_api.rb', 'lib/pulpcore_client/api/importers_pulp_api.rb', 'lib/pulpcore_client/api/importers_pulp_import_check_api.rb', 'lib/pulpcore_client/api/importers_pulp_imports_api.rb', 'lib/pulpcore_client/api/orphans_api.rb', 'lib/pulpcore_client/api/orphans_cleanup_api.rb', 'lib/pulpcore_client/api/publications_api.rb', 'lib/pulpcore_client/api/remotes_api.rb', 'lib/pulpcore_client/api/repair_api.rb', 'lib/pulpcore_client/api/repositories_api.rb', 'lib/pulpcore_client/api/repositories_reclaim_space_api.rb', 'lib/pulpcore_client/api/repository_versions_api.rb', 'lib/pulpcore_client/api/roles_api.rb', 'lib/pulpcore_client/api/signing_services_api.rb', 'lib/pulpcore_client/api/status_api.rb', 'lib/pulpcore_client/api/task_groups_api.rb', 'lib/pulpcore_client/api/task_schedules_api.rb', 'lib/pulpcore_client/api/tasks_api.rb', 'lib/pulpcore_client/api/uploads_api.rb', 'lib/pulpcore_client/api/upstream_pulps_api.rb', 'lib/pulpcore_client/api/users_api.rb', 'lib/pulpcore_client/api/users_roles_api.rb', 'lib/pulpcore_client/api/workers_api.rb', 'lib/pulpcore_client/api_client.rb', 'lib/pulpcore_client/api_error.rb', 'lib/pulpcore_client/configuration.rb', 'lib/pulpcore_client/models/access_policy.rb', 'lib/pulpcore_client/models/access_policy_response.rb', 'lib/pulpcore_client/models/artifact.rb', 'lib/pulpcore_client/models/artifact_distribution_response.rb', 'lib/pulpcore_client/models/artifact_response.rb', 'lib/pulpcore_client/models/async_operation_response.rb', 'lib/pulpcore_client/models/content_app_status_response.rb', 'lib/pulpcore_client/models/content_guard_response.rb', 'lib/pulpcore_client/models/content_redirect_content_guard.rb', 'lib/pulpcore_client/models/content_redirect_content_guard_response.rb', 'lib/pulpcore_client/models/content_settings_response.rb', 'lib/pulpcore_client/models/content_summary_response.rb', 'lib/pulpcore_client/models/database_connection_response.rb', 'lib/pulpcore_client/models/distribution_response.rb', 'lib/pulpcore_client/models/domain.rb', 'lib/pulpcore_client/models/domain_response.rb', 'lib/pulpcore_client/models/evaluation_response.rb', 'lib/pulpcore_client/models/filesystem_export.rb', 'lib/pulpcore_client/models/filesystem_export_response.rb', 'lib/pulpcore_client/models/filesystem_exporter.rb', 'lib/pulpcore_client/models/filesystem_exporter_response.rb', 'lib/pulpcore_client/models/group.rb', 'lib/pulpcore_client/models/group_progress_report_response.rb', 'lib/pulpcore_client/models/group_response.rb', 'lib/pulpcore_client/models/group_role.rb', 'lib/pulpcore_client/models/group_role_response.rb', 'lib/pulpcore_client/models/group_user.rb', 'lib/pulpcore_client/models/group_user_response.rb', 'lib/pulpcore_client/models/import_response.rb', 'lib/pulpcore_client/models/method_enum.rb', 'lib/pulpcore_client/models/minimal_task_response.rb', 'lib/pulpcore_client/models/multiple_artifact_content_response.rb', 'lib/pulpcore_client/models/my_permissions_response.rb', 'lib/pulpcore_client/models/nested_role.rb', 'lib/pulpcore_client/models/nested_role_response.rb', 'lib/pulpcore_client/models/object_roles_response.rb', 'lib/pulpcore_client/models/orphans_cleanup.rb', 'lib/pulpcore_client/models/paginated_access_policy_response_list.rb', 'lib/pulpcore_client/models/paginated_artifact_distribution_response_list.rb', 'lib/pulpcore_client/models/paginated_artifact_response_list.rb', 'lib/pulpcore_client/models/paginated_content_guard_response_list.rb', 'lib/pulpcore_client/models/paginated_content_redirect_content_guard_response_list.rb', 'lib/pulpcore_client/models/paginated_distribution_response_list.rb', 'lib/pulpcore_client/models/paginated_domain_response_list.rb', 'lib/pulpcore_client/models/paginated_filesystem_export_response_list.rb', 'lib/pulpcore_client/models/paginated_filesystem_exporter_response_list.rb', 'lib/pulpcore_client/models/paginated_group_response_list.rb', 'lib/pulpcore_client/models/paginated_group_role_response_list.rb', 'lib/pulpcore_client/models/paginated_group_user_response_list.rb', 'lib/pulpcore_client/models/paginated_import_response_list.rb', 'lib/pulpcore_client/models/paginated_multiple_artifact_content_response_list.rb', 'lib/pulpcore_client/models/paginated_publication_response_list.rb', 'lib/pulpcore_client/models/paginated_pulp_export_response_list.rb', 'lib/pulpcore_client/models/paginated_pulp_exporter_response_list.rb', 'lib/pulpcore_client/models/paginated_pulp_importer_response_list.rb', 'lib/pulpcore_client/models/paginated_rbac_content_guard_response_list.rb', 'lib/pulpcore_client/models/paginated_remote_response_list.rb', 'lib/pulpcore_client/models/paginated_repository_response_list.rb', 'lib/pulpcore_client/models/paginated_repository_version_response_list.rb', 'lib/pulpcore_client/models/paginated_role_response_list.rb', 'lib/pulpcore_client/models/paginated_signing_service_response_list.rb', 'lib/pulpcore_client/models/paginated_task_group_response_list.rb', 'lib/pulpcore_client/models/paginated_task_response_list.rb', 'lib/pulpcore_client/models/paginated_task_schedule_response_list.rb', 'lib/pulpcore_client/models/paginated_upload_response_list.rb', 'lib/pulpcore_client/models/paginated_upstream_pulp_response_list.rb', 'lib/pulpcore_client/models/paginated_user_response_list.rb', 'lib/pulpcore_client/models/paginated_user_role_response_list.rb', 'lib/pulpcore_client/models/paginated_worker_response_list.rb', 'lib/pulpcore_client/models/patched_access_policy.rb', 'lib/pulpcore_client/models/patched_content_redirect_content_guard.rb', 'lib/pulpcore_client/models/patched_domain.rb', 'lib/pulpcore_client/models/patched_filesystem_exporter.rb', 'lib/pulpcore_client/models/patched_group.rb', 'lib/pulpcore_client/models/patched_pulp_exporter.rb', 'lib/pulpcore_client/models/patched_pulp_importer.rb', 'lib/pulpcore_client/models/patched_rbac_content_guard.rb', 'lib/pulpcore_client/models/patched_role.rb', 'lib/pulpcore_client/models/patched_task_cancel.rb', 'lib/pulpcore_client/models/patched_upstream_pulp.rb', 'lib/pulpcore_client/models/patched_user.rb', 'lib/pulpcore_client/models/policy_enum.rb', 'lib/pulpcore_client/models/progress_report_response.rb', 'lib/pulpcore_client/models/publication_response.rb', 'lib/pulpcore_client/models/pulp_export.rb', 'lib/pulpcore_client/models/pulp_export_response.rb', 'lib/pulpcore_client/models/pulp_exporter.rb', 'lib/pulpcore_client/models/pulp_exporter_response.rb', 'lib/pulpcore_client/models/pulp_import.rb', 'lib/pulpcore_client/models/pulp_import_check.rb', 'lib/pulpcore_client/models/pulp_import_check_response.rb', 'lib/pulpcore_client/models/pulp_importer.rb', 'lib/pulpcore_client/models/pulp_importer_response.rb', 'lib/pulpcore_client/models/purge.rb', 'lib/pulpcore_client/models/rbac_content_guard.rb', 'lib/pulpcore_client/models/rbac_content_guard_response.rb', 'lib/pulpcore_client/models/reclaim_space.rb', 'lib/pulpcore_client/models/redis_connection_response.rb', 'lib/pulpcore_client/models/remote_response.rb', 'lib/pulpcore_client/models/remote_response_hidden_fields.rb', 'lib/pulpcore_client/models/repair.rb', 'lib/pulpcore_client/models/repository_response.rb', 'lib/pulpcore_client/models/repository_version_response.rb', 'lib/pulpcore_client/models/role.rb', 'lib/pulpcore_client/models/role_response.rb', 'lib/pulpcore_client/models/signing_service_response.rb', 'lib/pulpcore_client/models/states_enum.rb', 'lib/pulpcore_client/models/status_response.rb', 'lib/pulpcore_client/models/storage_class_enum.rb', 'lib/pulpcore_client/models/storage_response.rb', 'lib/pulpcore_client/models/task_group_operation_response.rb', 'lib/pulpcore_client/models/task_group_response.rb', 'lib/pulpcore_client/models/task_response.rb', 'lib/pulpcore_client/models/task_schedule_response.rb', 'lib/pulpcore_client/models/upload.rb', 'lib/pulpcore_client/models/upload_chunk.rb', 'lib/pulpcore_client/models/upload_chunk_response.rb', 'lib/pulpcore_client/models/upload_commit.rb', 'lib/pulpcore_client/models/upload_detail_response.rb', 'lib/pulpcore_client/models/upload_response.rb', 'lib/pulpcore_client/models/upstream_pulp.rb', 'lib/pulpcore_client/models/upstream_pulp_response.rb', 'lib/pulpcore_client/models/user.rb', 'lib/pulpcore_client/models/user_group.rb', 'lib/pulpcore_client/models/user_group_response.rb', 'lib/pulpcore_client/models/user_response.rb', 'lib/pulpcore_client/models/user_role.rb', 'lib/pulpcore_client/models/user_role_response.rb', 'lib/pulpcore_client/models/version_response.rb', 'lib/pulpcore_client/models/worker_response.rb', 'lib/pulpcore_client/version.rb', 'pulpcore_client.gemspec', 'spec/api/access_policies_api_spec.rb', 'spec/api/artifacts_api_spec.rb', 'spec/api/content_api_spec.rb', 'spec/api/contentguards_api_spec.rb', 'spec/api/contentguards_content_redirect_api_spec.rb', 'spec/api/contentguards_rbac_api_spec.rb', 'spec/api/distributions_api_spec.rb', 'spec/api/distributions_artifacts_api_spec.rb', 'spec/api/domains_api_spec.rb', 'spec/api/exporters_filesystem_api_spec.rb', 'spec/api/exporters_filesystem_exports_api_spec.rb', 'spec/api/exporters_pulp_api_spec.rb', 'spec/api/exporters_pulp_exports_api_spec.rb', 'spec/api/groups_api_spec.rb', 'spec/api/groups_roles_api_spec.rb', 'spec/api/groups_users_api_spec.rb', 'spec/api/importers_pulp_api_spec.rb', 'spec/api/importers_pulp_import_check_api_spec.rb', 'spec/api/importers_pulp_imports_api_spec.rb', 'spec/api/orphans_api_spec.rb', 'spec/api/orphans_cleanup_api_spec.rb', 'spec/api/publications_api_spec.rb', 'spec/api/remotes_api_spec.rb', 'spec/api/repair_api_spec.rb', 'spec/api/repositories_api_spec.rb', 'spec/api/repositories_reclaim_space_api_spec.rb', 'spec/api/repository_versions_api_spec.rb', 'spec/api/roles_api_spec.rb', 'spec/api/signing_services_api_spec.rb', 'spec/api/status_api_spec.rb', 'spec/api/task_groups_api_spec.rb', 'spec/api/task_schedules_api_spec.rb', 'spec/api/tasks_api_spec.rb', 'spec/api/uploads_api_spec.rb', 'spec/api/upstream_pulps_api_spec.rb', 'spec/api/users_api_spec.rb', 'spec/api/users_roles_api_spec.rb', 'spec/api/workers_api_spec.rb', 'spec/api_client_spec.rb', 'spec/configuration_spec.rb', 'spec/models/access_policy_response_spec.rb', 'spec/models/access_policy_spec.rb', 'spec/models/artifact_distribution_response_spec.rb', 'spec/models/artifact_response_spec.rb', 'spec/models/artifact_spec.rb', 'spec/models/async_operation_response_spec.rb', 'spec/models/content_app_status_response_spec.rb', 'spec/models/content_guard_response_spec.rb', 'spec/models/content_redirect_content_guard_response_spec.rb', 'spec/models/content_redirect_content_guard_spec.rb', 'spec/models/content_settings_response_spec.rb', 'spec/models/content_summary_response_spec.rb', 'spec/models/database_connection_response_spec.rb', 'spec/models/distribution_response_spec.rb', 'spec/models/domain_response_spec.rb', 'spec/models/domain_spec.rb', 'spec/models/evaluation_response_spec.rb', 'spec/models/filesystem_export_response_spec.rb', 'spec/models/filesystem_export_spec.rb', 'spec/models/filesystem_exporter_response_spec.rb', 'spec/models/filesystem_exporter_spec.rb', 'spec/models/group_progress_report_response_spec.rb', 'spec/models/group_response_spec.rb', 'spec/models/group_role_response_spec.rb', 'spec/models/group_role_spec.rb', 'spec/models/group_spec.rb', 'spec/models/group_user_response_spec.rb', 'spec/models/group_user_spec.rb', 'spec/models/import_response_spec.rb', 'spec/models/method_enum_spec.rb', 'spec/models/minimal_task_response_spec.rb', 'spec/models/multiple_artifact_content_response_spec.rb', 'spec/models/my_permissions_response_spec.rb', 'spec/models/nested_role_response_spec.rb', 'spec/models/nested_role_spec.rb', 'spec/models/object_roles_response_spec.rb', 'spec/models/orphans_cleanup_spec.rb', 'spec/models/paginated_access_policy_response_list_spec.rb', 'spec/models/paginated_artifact_distribution_response_list_spec.rb', 'spec/models/paginated_artifact_response_list_spec.rb', 'spec/models/paginated_content_guard_response_list_spec.rb', 'spec/models/paginated_content_redirect_content_guard_response_list_spec.rb', 'spec/models/paginated_distribution_response_list_spec.rb', 'spec/models/paginated_domain_response_list_spec.rb', 'spec/models/paginated_filesystem_export_response_list_spec.rb', 'spec/models/paginated_filesystem_exporter_response_list_spec.rb', 'spec/models/paginated_group_response_list_spec.rb', 'spec/models/paginated_group_role_response_list_spec.rb', 'spec/models/paginated_group_user_response_list_spec.rb', 'spec/models/paginated_import_response_list_spec.rb', 'spec/models/paginated_multiple_artifact_content_response_list_spec.rb', 'spec/models/paginated_publication_response_list_spec.rb', 'spec/models/paginated_pulp_export_response_list_spec.rb', 'spec/models/paginated_pulp_exporter_response_list_spec.rb', 'spec/models/paginated_pulp_importer_response_list_spec.rb', 'spec/models/paginated_rbac_content_guard_response_list_spec.rb', 'spec/models/paginated_remote_response_list_spec.rb', 'spec/models/paginated_repository_response_list_spec.rb', 'spec/models/paginated_repository_version_response_list_spec.rb', 'spec/models/paginated_role_response_list_spec.rb', 'spec/models/paginated_signing_service_response_list_spec.rb', 'spec/models/paginated_task_group_response_list_spec.rb', 'spec/models/paginated_task_response_list_spec.rb', 'spec/models/paginated_task_schedule_response_list_spec.rb', 'spec/models/paginated_upload_response_list_spec.rb', 'spec/models/paginated_upstream_pulp_response_list_spec.rb', 'spec/models/paginated_user_response_list_spec.rb', 'spec/models/paginated_user_role_response_list_spec.rb', 'spec/models/paginated_worker_response_list_spec.rb', 'spec/models/patched_access_policy_spec.rb', 'spec/models/patched_content_redirect_content_guard_spec.rb', 'spec/models/patched_domain_spec.rb', 'spec/models/patched_filesystem_exporter_spec.rb', 'spec/models/patched_group_spec.rb', 'spec/models/patched_pulp_exporter_spec.rb', 'spec/models/patched_pulp_importer_spec.rb', 'spec/models/patched_rbac_content_guard_spec.rb', 'spec/models/patched_role_spec.rb', 'spec/models/patched_task_cancel_spec.rb', 'spec/models/patched_upstream_pulp_spec.rb', 'spec/models/patched_user_spec.rb', 'spec/models/policy_enum_spec.rb', 'spec/models/progress_report_response_spec.rb', 'spec/models/publication_response_spec.rb', 'spec/models/pulp_export_response_spec.rb', 'spec/models/pulp_export_spec.rb', 'spec/models/pulp_exporter_response_spec.rb', 'spec/models/pulp_exporter_spec.rb', 'spec/models/pulp_import_check_response_spec.rb', 'spec/models/pulp_import_check_spec.rb', 'spec/models/pulp_import_spec.rb', 'spec/models/pulp_importer_response_spec.rb', 'spec/models/pulp_importer_spec.rb', 'spec/models/purge_spec.rb', 'spec/models/rbac_content_guard_response_spec.rb', 'spec/models/rbac_content_guard_spec.rb', 'spec/models/reclaim_space_spec.rb', 'spec/models/redis_connection_response_spec.rb', 'spec/models/remote_response_hidden_fields_spec.rb', 'spec/models/remote_response_spec.rb', 'spec/models/repair_spec.rb', 'spec/models/repository_response_spec.rb', 'spec/models/repository_version_response_spec.rb', 'spec/models/role_response_spec.rb', 'spec/models/role_spec.rb', 'spec/models/signing_service_response_spec.rb', 'spec/models/states_enum_spec.rb', 'spec/models/status_response_spec.rb', 'spec/models/storage_class_enum_spec.rb', 'spec/models/storage_response_spec.rb', 'spec/models/task_group_operation_response_spec.rb', 'spec/models/task_group_response_spec.rb', 'spec/models/task_response_spec.rb', 'spec/models/task_schedule_response_spec.rb', 'spec/models/upload_chunk_response_spec.rb', 'spec/models/upload_chunk_spec.rb', 'spec/models/upload_commit_spec.rb', 'spec/models/upload_detail_response_spec.rb', 'spec/models/upload_response_spec.rb', 'spec/models/upload_spec.rb', 'spec/models/upstream_pulp_response_spec.rb', 'spec/models/upstream_pulp_spec.rb', 'spec/models/user_group_response_spec.rb', 'spec/models/user_group_spec.rb', 'spec/models/user_response_spec.rb', 'spec/models/user_role_response_spec.rb', 'spec/models/user_role_spec.rb', 'spec/models/user_spec.rb', 'spec/models/version_response_spec.rb', 'spec/models/worker_response_spec.rb', 'spec/spec_helper.rb'], 'homepage': 'https://github.com/pulp/pulpcore', 'licenses': ['GPLv2+'], 'metadata': {}, 'post_install_message': None, 'rdoc_options': [], 'require_paths': ['lib'], 'required_ruby_version': UsrMarshal:Gem::Requirement({'requirements': [['>=', UsrMarshal:Gem::Version({'version': '1.9'})]]}), 'required_rubygems_version': UsrMarshal:Gem::Requirement({'requirements': [['>=', UsrMarshal:Gem::Version({'version': '0'})]]}), 'requirements': [], 'rubygems_version': '3.0.3.1', 'signing_key': None, 'specification_version': 4, 'summary': 'Pulp 3 API Ruby Gem', 'test_files': ['spec/api/tasks_api_spec.rb', 'spec/api/contentguards_content_redirect_api_spec.rb', 'spec/api/domains_api_spec.rb', 'spec/api/content_api_spec.rb', 'spec/api/exporters_pulp_exports_api_spec.rb', 'spec/api/upstream_pulps_api_spec.rb', 'spec/api/publications_api_spec.rb', 'spec/api/orphans_api_spec.rb', 'spec/api/uploads_api_spec.rb', 'spec/api/importers_pulp_api_spec.rb', 'spec/api/contentguards_rbac_api_spec.rb', 'spec/api/distributions_api_spec.rb', 'spec/api/orphans_cleanup_api_spec.rb', 'spec/api/roles_api_spec.rb', 'spec/api/task_schedules_api_spec.rb', 'spec/api/groups_users_api_spec.rb', 'spec/api/exporters_pulp_api_spec.rb', 'spec/api/groups_api_spec.rb', 'spec/api/repositories_reclaim_space_api_spec.rb', 'spec/api/groups_roles_api_spec.rb', 'spec/api/distributions_artifacts_api_spec.rb', 'spec/api/signing_services_api_spec.rb', 'spec/api/users_api_spec.rb', 'spec/api/task_groups_api_spec.rb', 'spec/api/status_api_spec.rb', 'spec/api/artifacts_api_spec.rb', 'spec/api/users_roles_api_spec.rb', 'spec/api/exporters_filesystem_exports_api_spec.rb', 'spec/api/contentguards_api_spec.rb', 'spec/api/repositories_api_spec.rb', 'spec/api/importers_pulp_imports_api_spec.rb', 'spec/api/importers_pulp_import_check_api_spec.rb', 'spec/api/repair_api_spec.rb', 'spec/api/exporters_filesystem_api_spec.rb', 'spec/api/access_policies_api_spec.rb', 'spec/api/workers_api_spec.rb', 'spec/api/remotes_api_spec.rb', 'spec/api/repository_versions_api_spec.rb', 'spec/api_client_spec.rb', 'spec/configuration_spec.rb', 'spec/models/paginated_worker_response_list_spec.rb', 'spec/models/remote_response_spec.rb', 'spec/models/paginated_multiple_artifact_content_response_list_spec.rb', 'spec/models/nested_role_spec.rb', 'spec/models/paginated_access_policy_response_list_spec.rb', 'spec/models/content_app_status_response_spec.rb', 'spec/models/paginated_pulp_exporter_response_list_spec.rb', 'spec/models/paginated_upstream_pulp_response_list_spec.rb', 'spec/models/artifact_response_spec.rb', 'spec/models/repair_spec.rb', 'spec/models/upload_commit_spec.rb', 'spec/models/publication_response_spec.rb', 'spec/models/group_role_spec.rb', 'spec/models/filesystem_export_response_spec.rb', 'spec/models/repository_version_response_spec.rb', 'spec/models/paginated_pulp_importer_response_list_spec.rb', 'spec/models/paginated_task_group_response_list_spec.rb', 'spec/models/patched_task_cancel_spec.rb', 'spec/models/paginated_repository_response_list_spec.rb', 'spec/models/paginated_publication_response_list_spec.rb', 'spec/models/worker_response_spec.rb', 'spec/models/patched_filesystem_exporter_spec.rb', 'spec/models/filesystem_export_spec.rb', 'spec/models/group_spec.rb', 'spec/models/filesystem_exporter_response_spec.rb', 'spec/models/upload_response_spec.rb', 'spec/models/domain_spec.rb', 'spec/models/redis_connection_response_spec.rb', 'spec/models/paginated_domain_response_list_spec.rb', 'spec/models/patched_content_redirect_content_guard_spec.rb', 'spec/models/patched_role_spec.rb', 'spec/models/status_response_spec.rb', 'spec/models/artifact_spec.rb', 'spec/models/patched_access_policy_spec.rb', 'spec/models/my_permissions_response_spec.rb', 'spec/models/user_spec.rb', 'spec/models/paginated_task_schedule_response_list_spec.rb', 'spec/models/group_response_spec.rb', 'spec/models/rbac_content_guard_response_spec.rb', 'spec/models/patched_user_spec.rb', 'spec/models/pulp_import_spec.rb', 'spec/models/task_response_spec.rb', 'spec/models/upload_detail_response_spec.rb', 'spec/models/filesystem_exporter_spec.rb', 'spec/models/group_user_response_spec.rb', 'spec/models/pulp_import_check_response_spec.rb', 'spec/models/paginated_rbac_content_guard_response_list_spec.rb', 'spec/models/group_user_spec.rb', 'spec/models/upstream_pulp_spec.rb', 'spec/models/pulp_importer_spec.rb', 'spec/models/group_progress_report_response_spec.rb', 'spec/models/role_response_spec.rb', 'spec/models/pulp_exporter_spec.rb', 'spec/models/paginated_repository_version_response_list_spec.rb', 'spec/models/paginated_filesystem_export_response_list_spec.rb', 'spec/models/method_enum_spec.rb', 'spec/models/distribution_response_spec.rb', 'spec/models/purge_spec.rb', 'spec/models/paginated_signing_service_response_list_spec.rb', 'spec/models/paginated_upload_response_list_spec.rb', 'spec/models/access_policy_spec.rb', 'spec/models/artifact_distribution_response_spec.rb', 'spec/models/group_role_response_spec.rb', 'spec/models/pulp_exporter_response_spec.rb', 'spec/models/policy_enum_spec.rb', 'spec/models/task_group_operation_response_spec.rb', 'spec/models/user_role_response_spec.rb', 'spec/models/patched_group_spec.rb', 'spec/models/remote_response_hidden_fields_spec.rb', 'spec/models/patched_domain_spec.rb', 'spec/models/minimal_task_response_spec.rb', 'spec/models/paginated_user_response_list_spec.rb', 'spec/models/upload_spec.rb', 'spec/models/content_settings_response_spec.rb', 'spec/models/patched_upstream_pulp_spec.rb', 'spec/models/access_policy_response_spec.rb', 'spec/models/reclaim_space_spec.rb', 'spec/models/storage_response_spec.rb', 'spec/models/evaluation_response_spec.rb', 'spec/models/paginated_content_guard_response_list_spec.rb', 'spec/models/paginated_content_redirect_content_guard_response_list_spec.rb', 'spec/models/paginated_group_response_list_spec.rb', 'spec/models/multiple_artifact_content_response_spec.rb', 'spec/models/paginated_artifact_response_list_spec.rb', 'spec/models/content_guard_response_spec.rb', 'spec/models/content_redirect_content_guard_response_spec.rb', 'spec/models/states_enum_spec.rb', 'spec/models/orphans_cleanup_spec.rb', 'spec/models/domain_response_spec.rb', 'spec/models/paginated_group_role_response_list_spec.rb', 'spec/models/signing_service_response_spec.rb', 'spec/models/content_summary_response_spec.rb', 'spec/models/object_roles_response_spec.rb', 'spec/models/patched_rbac_content_guard_spec.rb', 'spec/models/paginated_group_user_response_list_spec.rb', 'spec/models/upload_chunk_response_spec.rb', 'spec/models/paginated_role_response_list_spec.rb', 'spec/models/import_response_spec.rb', 'spec/models/paginated_distribution_response_list_spec.rb', 'spec/models/rbac_content_guard_spec.rb', 'spec/models/paginated_artifact_distribution_response_list_spec.rb', 'spec/models/user_response_spec.rb', 'spec/models/pulp_importer_response_spec.rb', 'spec/models/pulp_export_response_spec.rb', 'spec/models/storage_class_enum_spec.rb', 'spec/models/user_group_response_spec.rb', 'spec/models/patched_pulp_exporter_spec.rb', 'spec/models/database_connection_response_spec.rb', 'spec/models/version_response_spec.rb', 'spec/models/role_spec.rb', 'spec/models/content_redirect_content_guard_spec.rb', 'spec/models/repository_response_spec.rb', 'spec/models/pulp_export_spec.rb', 'spec/models/paginated_remote_response_list_spec.rb', 'spec/models/upload_chunk_spec.rb', 'spec/models/progress_report_response_spec.rb', 'spec/models/paginated_filesystem_exporter_response_list_spec.rb', 'spec/models/task_group_response_spec.rb', 'spec/models/patched_pulp_importer_spec.rb', 'spec/models/paginated_pulp_export_response_list_spec.rb', 'spec/models/nested_role_response_spec.rb', 'spec/models/pulp_import_check_spec.rb', 'spec/models/async_operation_response_spec.rb', 'spec/models/paginated_user_role_response_list_spec.rb', 'spec/models/user_group_spec.rb', 'spec/models/paginated_task_response_list_spec.rb', 'spec/models/upstream_pulp_response_spec.rb', 'spec/models/task_schedule_response_spec.rb', 'spec/models/user_role_spec.rb', 'spec/models/paginated_import_response_list_spec.rb', 'spec/spec_helper.rb']})
```